{"assignees":[],"author":{"id":"MDQ6VXNlcjIwNDYxMDEz","is_bot":false,"login":"rjzamora","name":"Richard (Rick) Zamora"},"body":"**Describe the bug**\r\nFor cumulative operations (e.g. `cummax`, `cummin`, `cumsum`, `cumprod`), Pandas defines `axis` and `skipna` as optional positional arguments, while cudf requires these to be key-word arguments.\r\n\r\n**Steps/Code to reproduce bug**\r\n```python\r\nimport cudf\r\nimport pandas as pd\r\n\r\ndf = pd.DataFrame({\"a\": [1, 2]})\r\naxis = 0\r\nskipna = False\r\ndf.cumsum(axis, skipna)\r\n```\r\n```\r\n   a\r\n0  1\r\n1  3\r\n```\r\n```python\r\ncudf.from_pandas(df).cumsum(axis, skipna)\r\n```\r\n```\r\n---------------------------------------------------------------------------\r\nTypeError                                 Traceback (most recent call last)\r\nCell In[34], line 7\r\n      5 axis = 0\r\n      6 skipna = False\r\n----> 7 cudf.from_pandas(df).cumsum(axis, skipna)\r\n\r\nFile /.../site-packages/cudf/core/mixins/mixin_factory.py:11, in _partialmethod.<locals>.wrapper(self, *args2, **kwargs2)\r\n     10 def wrapper(self, *args2, **kwargs2):\r\n---> 11     return method(self, *args1, *args2, **kwargs1, **kwargs2)\r\n\r\nFile /.../site-packages/nvtx/nvtx.py:115, in annotate.__call__.<locals>.inner(*args, **kwargs)\r\n    112 @wraps(func)\r\n    113 def inner(*args, **kwargs):\r\n    114     libnvtx_push_range(self.attributes, self.domain.handle)\r\n--> 115     result = func(*args, **kwargs)\r\n    116     libnvtx_pop_range(self.domain.handle)\r\n    117     return result\r\n\r\nTypeError: DataFrame._scan() got multiple values for argument 'op'\r\n```\r\n\r\n**Expected behavior**\r\nI'd expect cudf behavior to be consistent with pandas in this case.","closed":false,"closedAt":null,"comments":[],"createdAt":"2024-02-28T19:01:04Z","id":"I_kwDOBWUGps6AuZ3D","labels":[{"id":"MDU6TGFiZWw1OTk2MjY1NTk=","name":"bug","description":"Something isn't working","color":"d73a4a"}],"milestone":null,"number":15176,"projectCards":[],"projectItems":[],"reactionGroups":[],"state":"OPEN","title":"[BUG] Discrepancy between cudf and pandas for cumulative operations","updatedAt":"2024-02-28T19:01:04Z","url":"https://github.com/rapidsai/cudf/issues/15176"}
